### DISCUSSION TITLE
query about batchsize

### DISCUSSION THREAD

**Discussion by 0xMALVEE** [bbc/sqs-consumer#502] [Help] [answered]
### Summary

hey, 
If I set batchSize more then 1 let's say 10, do I have to use the handleMessageBatch method or handleMessage would also work and would run all message togehter? 
Thanks

**Comment by nicholasgriffintn** [comment]
The main difference between the two is that handleMessage will receive individual messages, in parallel, whereas handleMessageBatch will receive the full messages array.

**Reply by nicholasgriffintn** [reply]
(Also that the equivalent AWS SDK calls will be made when using batch, so deletes, etc will be done as a batch, rather than individually)

**Reply by 0xMALVEE** [reply]
so the handleMessage is automaticly getting the messages array and executing in parallel right?

**Reply by nicholasgriffintn** [reply, marked as answer]
handleMessage will only be sent each individual message and executed individually per message.

handleMessageBatch will be sent the full array and executed with that full array.

**Reply by 0xMALVEE** [reply]
btw how does it determine when to get next batch?

**Reply by nicholasgriffintn** [reply]
This is configured via pollingWaitTimeMs: https://bbc.github.io/sqs-consumer/interfaces/ConsumerOptions.html#pollingWaitTimeMs

And waitTimeSeconds: https://bbc.github.io/sqs-consumer/interfaces/ConsumerOptions.html#waitTimeSeconds

pollingWaitTimeMs is the amount of time in between the handlers returning an acknowledgement for all messages and the next polling interval.

waitTimeSeconds is the AWS SQS setting, which is the amount of time to wait for messages before returning from sqs.
